src/components/organisms/SearchPane/SearchPane.tsx

```diff
@@ -0,0 +1,524 @@
+import {ipcRenderer} from 'electron';
+
+import React, {Key, useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useSelector} from 'react-redux';
+import {useUpdateEffect} from 'react-use';
+
+import {Input, Modal} from 'antd';
+
+import {ExclamationCircleOutlined} from '@ant-design/icons';
+
+import log from 'loglevel';
+import micromatch from 'micromatch';
+import path from 'path';
+
+import {DEFAULT_PANE_TITLE_HEIGHT, ROOT_FILE_ENTRY} from '@constants/constants';
+
+import {AlertEnum} from '@models/alert';
+import {FileEntry} from '@models/fileentry';
+
+import {useAppDispatch, useAppSelector} from '@redux/hooks';
+import {setAlert} from '@redux/reducers/alert';
+import {updateProjectConfig} from '@redux/reducers/appConfig';
+import {selectFile, setSelectingFile, updateResourceFilter} from '@redux/reducers/main';
+import {
+  openCreateFolderModal,
+  openNewResourceWizard,
+  openRenameEntityModal,
+  setExpandedFolders,
+} from '@redux/reducers/ui';
+import {fileIncludesSelector, isInPreviewModeSelector, scanExcludesSelector, settingsSelector} from '@redux/selectors';
+import {getResourcesForPath} from '@redux/services/fileEntry';
+import {getHelmValuesFile} from '@redux/services/helm';
+import {isKustomizationResource} from '@redux/services/kustomize';
+import {startPreview, stopPreview} from '@redux/services/preview';
+import {setRootFolder} from '@redux/thunks/setRootFolder';
+
+import {TitleBar} from '@molecules';
+
+import electronStore from '@utils/electronStore';
+import {uniqueArr} from '@utils/index';
+
+import {createNode} from '../FileTreePane/CreateNode';
+import TreeItem from '../FileTreePane/TreeItem';
+import {ProcessingEntity, TreeNode} from '../FileTreePane/types';
+import {createFilteredNode} from './CreateFilteredNode';
+import RecentSearch from './RecentSearch';
+
+import * as S from './styled';
+
+type Props = {
+  height: number;
+};
+
+type MatchParamProps = {
+  matchCase: boolean;
+  matchWholeWord: boolean;
+  regExp: boolean;
+};
+
+const SearchPane: React.FC<Props> = ({height}) => {
```

Comment:
This file is quite large, I would recommend extracting logic into separate files as custom hooks, by grouping relevant useState + useEffect and other hooks together in custom ones.

Reply:
did some refactoring for both SearchPane.tsx and FileTreePane.tsx, moved main chunks of logic to hooks/fileTreeHooks